src/source/raster_dem_tile_worker_source.js

```diff
@@ -1,31 +1,55 @@
 // @flow
 
 import DEMData from '../data/dem_data';
+import {RGBAImage} from '../util/image';
+import window from '../util/window';
 
 import type Actor from '../util/actor';
 import type {
     WorkerDEMTileParameters,
     WorkerDEMTileCallback,
     TileParameters
 } from './worker_source';
+const {ImageBitmap} = window;
 
 class RasterDEMTileWorkerSource {
     actor: Actor;
     loaded: {[string]: DEMData};
+    offscreenCanvas: OffscreenCanvas;
+    offscreenCanvasContext: CanvasRenderingContext2D;
 
     constructor() {
         this.loaded = {};
     }
 
     loadTile(params: WorkerDEMTileParameters, callback: WorkerDEMTileCallback) {
         const {uid, encoding, rawImageData} = params;
-        const dem = new DEMData(uid, rawImageData, encoding);
-
+        // Main thread will transfer ImageBitmap if offscreen decode with OffscreenCanvas is supported, else it will transfer an already decoded image.
+        const imagePixels = (ImageBitmap && rawImageData instanceof ImageBitmap) ? this.getImageData(rawImageData) : rawImageData;
+        const dem = new DEMData(uid, imagePixels, encoding);
         this.loaded = this.loaded || {};
         this.loaded[uid] = dem;
         callback(null, dem);
     }
 
+    getImageData(imgBitmap: ImageBitmap): RGBAImage {
+        // Lazily initialize OffscreenCanvas
+        if (!this.offscreenCanvas || !this.offscreenCanvasContext) {
+            // Dem tiles are typically 256x256
+            this.offscreenCanvas = new OffscreenCanvas(256, 256);
```

Comment:
```suggestion
            this.offscreenCanvas = new OffscreenCanvas(imgBitmap.width, imgBitmap.height);
```

They're resized below anyway.